### DISCUSSION TITLE
Disable nvim-cmp's <c-space> mapping

### DISCUSSION THREAD

**Discussion by mikker** [LazyVim/LazyVim#514] [Q&A] [answered]
How do I disable `nvim-cmp`'s mapping of `<c-space>`? I like vim's original behaviour better.

I've tried a few things but no luck yet. I think because `nvim-cmp` isn't loaded until `InsertEnter`?

https://www.lazyvim.org/plugins/coding#nvim-cmp for reference

Thanks!

**Comment by dpetka2001** [comment]
```lua
return { 
 {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = cmp.mapping.preset.insert(vim.tbl_deep_extend("force", opts.mapping, {
        ["<c-space>"] = cmp.config.disable,
      })),
    end,
  },
}
```
I think this should do it.

**Reply by kang8** [reply]
Cool! Before this, I have been using `vim.NIL` to disable it.

```diff
-        ["<c-space>"] = vim.NIL,
+        ["<c-space>"] = cmp.config.disable,
```

**Reply by mikker** [reply, marked as answer]
Actually, I can only make `vim.NIL` work

```lua
return { 
 {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = cmp.mapping.preset.insert(vim.tbl_deep_extend("force", opts.mapping, {
        ["<c-space>"] = vim.NIL,
      }))
    end,
  },
}
```

**Reply by dpetka2001** [reply]
Strange because `cmp.config.disable` sets the value to `vim.NIL` behind the scenes, so it should work either way. Whatever, since you got it working that's what matters.

**Reply by kang8** [reply]
> `cmp.config.disable` sets the value to `vim.NIL` behind the scenes

Yes, I think using `cmp.config.disable` to set it is a better way.
